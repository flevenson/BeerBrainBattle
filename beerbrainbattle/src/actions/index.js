export const addQuestion = (question) => ({
  type: 'ADD_QUESTION',
  question
})

export const addPlayers = (players) => ({
  type: 'ADD_PLAYERS',
  players
})

export const addVote = (answer) => ({
  type: 'ADD_VOTE',
  answer
})

export const filterPlayers = (players) => ({
  type: 'FILTER_PLAYERS',
  players
})

export const addPrize = (prize) => ({
  type: 'ADD_PRIZE',
  prize
})