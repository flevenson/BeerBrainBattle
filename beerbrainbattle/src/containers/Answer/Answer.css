.answer {
  height: 200px;
  width: 300px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-around;
  background-color: #C7EAF0;
  color: #2E2453;
  border: 2px solid #2E2453;
  border-radius: 5px;
  padding: 3px;
  font-size: 1.1rem;
  margin-top: 200px;
}