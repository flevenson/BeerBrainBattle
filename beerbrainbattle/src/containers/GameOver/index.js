import React from 'react';
import './GameOver.css';
import Proptypes from 'prop-types';
import { connect } from 'react-redux';
import { addPlayers } from '../../actions';
import { withRouter } from 'react-router';
import * as BeerData from '../../assets/BeerData.js';

export const GameOver = (props) => {

  if(!Object.keys(props.question).length){
    props.history.push('/')
  } 

  let correctAnswer

  if(Object.keys(props.question).length){
    correctAnswer = props.question.answers.find(answer => answer.correct === true)
  return(
    <div className='answer-container'>
      <div className='answer'>
        <h1 className='game-over'>GAME OVER</h1>
        <p className='correct-answer-title'>The Correct Answer Was:</p>
        <p className='correct-answer-answer'>{correctAnswer.answer}</p>
        <p className='award'>You Won {props.prize}</p>
      </div>
      <button 
        onClick={() => {
          props.addPlayers(0)
          props.history.push('/')
        }} 
        className='next-round'>Play Again!</button>
    </div>
    )
  } else {
    return (
      <div>
      </div>
    )
  }


}

export const mapStateToProps = (state) => ({
  players: state.players,
  question: state.question,
  prize: state.prize
})

export const mapDispatchToProps = (dispatch) => ({
  addPlayers: (players) => dispatch(addPlayers(players)),
})

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(GameOver))