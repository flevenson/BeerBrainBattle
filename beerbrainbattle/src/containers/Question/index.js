import React from 'react';
import './Question.css';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { withRouter } from 'react-router';
import { addVote } from '../../actions'

export const Question = (props) => {

  if(!Object.keys(props.question).length && !props.players){
    props.history.push('/')
  } 
  
  if(Object.keys(props.question).length){

  const answers = props.question.answers.map(answer => 
    <button 
      className='answer-button'
      onClick={() => props.addVote(answer.answer)}
      >{answer.answer}   {answer.numVotes}
    </button>
)

  const totalVotes = props.question.answers.reduce((allVotes, answer) => {
    allVotes += answer.numVotes
    return allVotes
  }, 0)

  console.log(totalVotes)

  if(totalVotes === parseInt(props.players) && Object.keys(props.question).length) {
    props.history.push('/answer')
  }


    return (
      <div className='question-holder'>
        <div className='question'>
          <h1>{props.question.question}</h1>
        </div>
        <div className='answers'>
          { answers }
        </div>
      </div>
      )
  }
  else {
    return (
      <div>
      </div>
      )
  }

}

export const mapStateToProps = (state) => ({
  players: state.players,
  question: state.question
})

export const mapDispatchToProps = (dispatch) => ({
  addVote: (answer) => dispatch(addVote(answer)),
})


export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Question))