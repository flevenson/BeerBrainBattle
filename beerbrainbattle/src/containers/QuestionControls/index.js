import React, { Component } from 'react';
import { connect } from 'react-redux';
import { addPlayers, addQuestion, addPrize } from '../../actions'
import './QuestionControls.css';
import PropTypes from 'prop-types';
import * as BeerData from '../../assets/BeerData.js';
import * as API from '../../utils'
import { withRouter } from 'react-router';

export class QuestionControls extends Component {
  constructor(props) {
    super(props)
    this.state = {
      category: '',
      difficulty: '',
      numPlayers: '',
      prize: '',
      showCategories: false,
      showDifficulty: false,
    }
  }


  toggleShowCategories = () => {
    this.setState({
      showCategories: !this.state.showCategories
    })
  }

  setCategory = (event) => {
    const category = event.target.innerText

    this.setState({
      category,
      showCategories: !this.state.showCategories
    })
  }

  toggleShowDifficulty = () => {
    this.setState({
      showDifficulty: !this.state.showDifficulty
    })
  }

  setDifficulty = (event) => {
    const difficulty = event.target.innerText

    this.setState({
      difficulty,
      showDifficulty: !this.state.showDifficulty
    })
  }  

  handleInputChange = (event) => {
    const { name, value } = event.target
    this.setState({
      [name]: value
    })
  }

  handleSubmit = async (event) => {
    event.preventDefault();
    const { players, category, difficulty, numPlayers, prize } = this.state
    if(this.props.players !== 0){
      this.setState({
        numPlayers: this.props.players
      })
    }
    const question = await API.fetchRandomQuestion(category, difficulty);
    this.props.addQuestion(question);
    this.props.addPlayers(numPlayers);
    this.makePrize();
    await this.props.history.push('/question')  
}

  makePrize = () => {
    if(!this.state.prize.length) {
      const randomBeer = BeerData.beerData[Math.floor(Math.random() * 100)]
      const randomPrize = `${randomBeer.year} ${randomBeer.beerName} from  ${randomBeer.brewery} worth $${randomBeer.value}`
      this.props.addPrize(randomPrize)
    } else {
      this.props.addPrize(this.state.prize)
    }
  }

  render() {

  const { showDifficulty, showCategories, category, difficulty, numPlayers, prize } = this.state;


  const categoryOptions = BeerData.categories.map(categoryData => 
    <li 
      className={ 
        showCategories 
        ? 'dropdown-option' 
        : 'hidden' } 
      onClick={ this.setCategory }>{ categoryData } 
    </li>
  )

    return(
      <form className='question-controls'>
        <div>
          <div className='dropdown-title' onClick={ this.toggleShowCategories}>{ category.length ? category : 'Category' }</div>
          <ul className='category-holder'>
            { categoryOptions }
          </ul>
        </div>
        <div>
          <div className='dropdown-title' onClick={ this.toggleShowDifficulty}>{ difficulty.length ? difficulty : 'Difficulty'}</div>
          <ul>
            <li 
              className={ 
                showDifficulty 
                ? 'dropdown-option' 
                : 'hidden'} 
              onClick={this.setDifficulty}>Easy
            </li>
            <li 
              className={ 
                showDifficulty 
                ? 'dropdown-option' 
                : 'hidden'} 
              onClick={this.setDifficulty}>Medium
            </li>
            <li 
              className={ 
                showDifficulty 
                ? 'dropdown-option' 
                : 'hidden'} 
              onClick={this.setDifficulty}>Hard
            </li>
          </ul>
        </div>
        <input 
          className={this.props.players ? 'hidden' : 'dropdown-title'} 
          placeholder='Number Of Players' 
          value={ numPlayers } 
          onChange={this.handleInputChange} 
          name='numPlayers'>
        </input>
        <input 
          className={this.props.prize ? 'hidden' : 'dropdown-title'} 
          placeholder='What will you Wager?' 
          value={ prize } 
          onChange={this.handleInputChange}
          name='prize'>
        </input>
        <button className='dropdown-title' onClick={this.handleSubmit}> Battle </button>
      </form>
    )
  }
}

export const mapStateToProps = (state) => ({
  players: state.players
})

export const mapDispatchToProps = (dispatch) => ({
  addPlayers: (players) => dispatch(addPlayers(players)),
  addQuestion: (question) => dispatch(addQuestion(question)),
  addPrize: (prize) => dispatch(addPrize(prize))
})

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(QuestionControls))