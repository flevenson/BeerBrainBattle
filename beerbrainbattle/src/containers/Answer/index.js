import React from 'react';
import './Answer.css';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { withRouter } from 'react-router';

export const Answer = (props) => {
  if(!Object.keys(props.question).length){
    props.history.push('/')
  } 

  let correctAnswer

  if(Object.keys(props.question).length){
    correctAnswer = props.question.answers.find(answer => answer.correct === true)
  return(
    <div>
      <p className='correct-answer-title'>The Correct Answer Is</p>
      <p>{correctAnswer.answer}</p>
    </div>
    )
  } else {
    return (
      <div>
      </div>
    )
  }

}

export const mapStateToProps = (state) => ({
  players: state.players,
  question: state.question
})

export default withRouter(connect(mapStateToProps)(Answer))