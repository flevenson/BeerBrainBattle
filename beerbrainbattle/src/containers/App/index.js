import React, { Component } from 'react';
import logo from '../../logo.svg';
import './App.css';
import PropTypes from 'prop-types';
import { Route, withRouter } from 'react-router-dom';
import QuestionControls from '../QuestionControls';
import Question from '../Question'
import Answer from '../Answer'
import { connect } from 'react-redux';

export const App = (props) => {

    return (
      <div className="App">
        <header>
          <h1 className='app-heading'>
            <span className='heading-beer'>Beer</span>
            <span className='heading-brain'>Brain</span>
            <span className='heading-battle'>Battle</span>
          </h1>
            <span className='player-count-holder'>
                <h1 className='player-count'>{props.players}</h1>
                <h1 className='players'>players</h1>
            </span>
        </header>
        <main className='app-main'>
          <Route 
            exact path='/'
            render={() => <QuestionControls />}
          />
          <Route 
            exact path='/question'
            render={() => <Question />}
          />
          <Route 
            exact path='/answer'
            render={() => <Answer />}
          />
        </main>
      </div>
    )

}

export const mapStateToProps = (state) => ({
  players: state.players
})

// export const mapDispatchToProps = (dispatch) => ({

// })

export default withRouter(connect(mapStateToProps)(App));
