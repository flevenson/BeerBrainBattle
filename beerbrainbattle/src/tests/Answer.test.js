import React from 'react';
import { shallow } from 'enzyme';
import { mockQuestion } from './testMocks.js'
import { Answer, mapStateToProps } from '../containers/Answer';
import * as Actions from '../actions'


describe('Answer', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Answer 
        question={mockQuestion[0]} />);
  })

  describe('Answer Component', () => {
    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })
})