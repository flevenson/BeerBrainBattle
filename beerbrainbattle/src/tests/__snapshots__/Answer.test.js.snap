// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Answer Answer Component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Answer
    question={
      Object {
        "answers": Array [
          Object {
            "answer": "5",
            "correct": false,
            "numVotes": 0,
          },
          Object {
            "answer": "2",
            "correct": false,
            "numVotes": 0,
          },
          Object {
            "answer": "4",
            "correct": true,
            "numVotes": 0,
          },
          Object {
            "answer": "0",
            "correct": false,
            "numVotes": 0,
          },
        ],
        "category": "Math",
        "difficulty": "Easy",
        "isCorrectlyAnswered": false,
        "question": "What is Two plus Two",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="answer"
        >
          <p
            className="correct-answer-title"
          >
            The Correct Answer Is:
          </p>
          <p
            className="correct-answer-answer"
          >
            4
          </p>
        </div>,
        <button
          className="next-round"
          onClick={[Function]}
        >
          Next Round
        </button>,
      ],
      "className": "answer-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="correct-answer-title"
            >
              The Correct Answer Is:
            </p>,
            <p
              className="correct-answer-answer"
            >
              4
            </p>,
          ],
          "className": "answer",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "The Correct Answer Is:",
              "className": "correct-answer-title",
            },
            "ref": null,
            "rendered": "The Correct Answer Is:",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "4",
              "className": "correct-answer-answer",
            },
            "ref": null,
            "rendered": "4",
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Next Round",
          "className": "next-round",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Next Round",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="answer"
          >
            <p
              className="correct-answer-title"
            >
              The Correct Answer Is:
            </p>
            <p
              className="correct-answer-answer"
            >
              4
            </p>
          </div>,
          <button
            className="next-round"
            onClick={[Function]}
          >
            Next Round
          </button>,
        ],
        "className": "answer-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="correct-answer-title"
              >
                The Correct Answer Is:
              </p>,
              <p
                className="correct-answer-answer"
              >
                4
              </p>,
            ],
            "className": "answer",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "The Correct Answer Is:",
                "className": "correct-answer-title",
              },
              "ref": null,
              "rendered": "The Correct Answer Is:",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "4",
                "className": "correct-answer-answer",
              },
              "ref": null,
              "rendered": "4",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Next Round",
            "className": "next-round",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Next Round",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
